commons/stan_files/banditNarm_2par_lapse.stan

```diff
@@ -0,0 +1,174 @@
+#include /pre/license.stan
+
+// Seymour et al 2012 J neuro model, w/o C (chioce perseveration) but with xi (lapse rate)
+// w/o reward sensitivity and punishment sensitivity
+// in sum, there are three parameters - Arew, Apun, xi
+// Aylward et al., 2018, PsyArXiv
+data {
+  int<lower=1> N;
+  int<lower=1> T;
+  int<lower=1, upper=T> Tsubj[N];
+  real rew[N, T];
+  real los[N, T];
+  int choice[N, T];
+  int Narm;
+}
+
+transformed data {
+  vector[Narm] initV;
+  initV = rep_vector(0.0, Narm);
+}
+
+parameters {
+  // Declare all parameters as vectors for vectorizing
+  // Hyper(group)-parameters
+  vector[3] mu_pr;
+  vector<lower=0>[3] sigma;
+
+  // Subject-level raw parameters (for Matt trick)
+  vector[N] Arew_pr;
+  vector[N] Apun_pr;
+  vector[N] xi_pr;
+}
+
+transformed parameters {
+  // Transform subject-level raw parameters
+  vector<lower=0, upper=1>[N] Arew;
+  vector<lower=0, upper=1>[N] Apun;
+  vector<lower=0, upper=1>[N] xi;
+
+  for (i in 1:N) {
+    Arew[i] = Phi_approx(mu_pr[1] + sigma[1] * Arew_pr[i]);
+    Apun[i] = Phi_approx(mu_pr[2] + sigma[2] * Apun_pr[i]);
+    xi[i]   = Phi_approx(mu_pr[3] + sigma[3] * xi_pr[i]);
+  }
+}
+
+model {
+  // Hyperparameters
+  mu_pr  ~ normal(0, 1);
+  sigma ~ normal(0, 0.2);
+
+  // individual parameters
+  Arew_pr  ~ normal(0, 1.0);
+  Apun_pr  ~ normal(0, 1.0);
+  xi_pr    ~ normal(0, 1.0);
+
+  for (i in 1:N) {
+    // Define values
+    vector[Narm] Qr;
+    vector[Narm] Qp;
+    vector[Narm] PEr_fic; // prediction error - for reward fictive updating (for unchosen options)
+    vector[Narm] PEp_fic; // prediction error - for punishment fictive updating (for unchosen options)
+    vector[Narm] Qsum;    // Qsum = Qrew + Qpun + perseverance
+
+    real Qr_chosen;
+    real Qp_chosen;
+    real PEr; // prediction error - for reward of the chosen option
+    real PEp; // prediction error - for punishment of the chosen option
+
+    // Initialize values
+    Qr    = initV;
+    Qp    = initV;
+    Qsum  = initV;
+
+    for (t in 1:Tsubj[i]) {
+      // softmax choice + xi (noise)
+      choice[i, t] ~ categorical(softmax(Qsum) * (1-xi[i]) + xi[i]/Narm);
+
+      // Prediction error signals
+      PEr     = rew[i, t] - Qr[choice[i, t]];
+      PEp     = los[i, t] - Qp[choice[i, t]];
+      PEr_fic = -Qr;
+      PEp_fic = -Qp;
+
+      // store chosen deck Q values (rew and pun)
+      Qr_chosen = Qr[choice[i, t]];
+      Qp_chosen = Qp[choice[i, t]];
+
+      // First, update Qr & Qp for all decks w/ fictive updating
+      Qr += Arew[i] * PEr_fic;
+      Qp += Apun[i] * PEp_fic;
+      // Replace Q values of chosen deck with correct values using stored values
+      Qr[choice[i, t]] = Qr_chosen + Arew[i] * PEr;
+      Qp[choice[i, t]] = Qp_chosen + Apun[i] * PEp;
+
+      // Q(sum)
+      Qsum = Qr + Qp;
+    }
+  }
```

Comment:
It's not necessary, but if available, I recommend you to pre-compute log likelihoods in the transformed parameters block and then add it to `target` in the model block.

By doing so, you can prevent computing the log likelihoods twice (in the model block and in the generated quantities block).

Reply:
I just pasted the original codes. I will revisit the issue later.